MTU should be bytes.

Sakaizd opened the pull request:
MTU should be bytes in ### QUIC Efficiency Percentage Calculation

the UDP packets should be (100000/1450).

BTW, may I ask how did you calculate the UDP Bytes emitted to be 104500 Bytes? I try to assume 1% of packet loss causing 101% of sending bytes, that is 101,000 Bytes that needs to send, and the total UDP Bytes should be (101000/1450)*1500=104,482

Sakaizd committed:
MTU should be bytes.

MTU should be bytes in ### QUIC Efficiency Percentage Calculation

the UDP packets should be (100000/1450).

BTW, may I ask how did you calculate the UDP Bytes emitted to be 104500 Bytes?
I try to assume 1% of packet loss causing 101% of sending bytes, that is 101,000 Bytes that needs to send, and the total UDP Bytes should be (101000/1450)*1500=104,482

Sparika commented:
Thanks for the PR. I will review proceed once I am back from vacations but it seems good.

Regarding the way of calculating the UDP bytes emitted for 1% of packet loss, your computation is correct. In my opinion this is here a simulation to exemplify what would be the QUIC efficiency. In a real example, you would not have a loss of 1% of bytes, or not even a loss of 1% of packets but a loss of some discrete number of packets. The total number of bytes transmitted in the example calculation depends on what is being considered for retransmission: 

(100000/1450) * 1500 * 1,01 = 104482 bytes (retransmission of 1% of all bytes)

69 * 1500 * 1,01 = 104535 bytes (retransmission of 1% of all packets)

70 * 1500 = 105000 bytes (retransmission of exactly 1 packet for a base of 69 packets)

I've used 104500 here as an approximation when I wrote the example, but I should make it cleared given your remark.

Sakaizd committed:
Enlarge the sending bytes to 1,000,000

Enlarge the sending bytes to 1,000,000 in ### QUIC Efficiency Percentage Calculation

Sakaizd commented:
Enjoy your vacation! 🎉🎉🎉

Thanks for the explanation, now I get it.

I enlarged the sending bytes to 1,000,000 and let the UDP packet number be 690, similar to the original number 552.
I think this can make it more clear to illustrate the extra packets that need to be sent.

Sparika commented:
Thanks for the fix and the update to the example. Appreciated.
